api/include/opentelemetry/logs/logger_type_traits.h

```diff
@@ -48,9 +48,22 @@ template <>
 struct LogRecordSetterTrait<EventId>
 {
   template <class ArgumentType>
-  inline static LogRecord *Set(LogRecord *log_record, ArgumentType && /*arg*/) noexcept
+  inline static LogRecord *Set(LogRecord *log_record, ArgumentType &&arg) noexcept
+  {
+    log_record->SetEventId(arg.id_, nostd::string_view{arg.name_.get()});
+
+    return log_record;
+  }
+};
+
+template <>
+struct LogRecordSetterTrait<int64_t>
```

Comment:
Is it conflicted with Body?it can also be converted from `int64_t` implicitly.

Reply:
Yes, thanks for the catch. Seems the template part of filling logging record requires each parameter to be in different type, and the `any` type on body make it conflict with other parameters. Do you have any suggestion for this problem?

Reply:
I have no idea for this problem by now, and I only avoided to specialize templates for these types before.
Maybe we can add a unit test to check the mistake.

Reply:
I removed the passing of int64_t to SDK, instead, a temporary EventId struct will be constructed which should fix the issue.

For unittest, do you mean to cover all the variant types of `body` field?

Reply:
Yes, use EmitLogRecord to cover all the variant types of body field.
Maybe we can add it in another PR later.

Reply:
FYI, I added an integration test to cover if the EventId passed to the user-facing API is passed correctly to the SDK.

https://github.com/ThomsonTan/opentelemetry-cpp/blob/otel_cpp_log_sdk_impl/exporters/ostream/test/ostream_log_test.cc#L465